opendebates/views.py

```diff
@@ -157,21 +164,13 @@ def vote(request, id):
         url = url + "#i"+str(idea.id)
         return redirect(url)
 
-    try:
-        related1 = Submission.objects.filter(category=idea.category,
-                                             duplicate_of__isnull=True,
-                                             approved=True).exclude(id=idea.id)[0]
-    except IndexError:
-        related1 = None
-    try:
-        related2 = Submission.objects.filter(category=idea.category,
-                                             duplicate_of__isnull=True,
-                                             approved=True).exclude(
-                                                 id=idea.id).exclude(id=related1.id)[0]
-    except (IndexError, AttributeError):  # if related1 is None -> related1.id
-        related2 = None
-
     if request.method == "GET":
+        two_other_approved_ideas = list(Submission.objects.filter(
+            category=idea.category,
+            duplicate_of=None,
+            approved=True).exclude(id=idea.id)[:2]) + [None, None]
+        related1 = two_other_approved_ideas[0]
+        related2 = two_other_approved_ideas[1]
```

Comment:
This is going to fetch and materialize many more submissions than it actually uses, can we avoid that? You could achieve similar with `itertools.chain(Submission..., [None, None])` and two calls of `next()`